test/extensions/filters/http/cdn_loop/parser_fuzz_test.cc

```diff
@@ -0,0 +1,19 @@
+#include "extensions/filters/http/cdn_loop/parser.h"
+
+#include "test/fuzz/fuzz_runner.h"
+
+#include "absl/strings/string_view.h"
+
+namespace Envoy {
+namespace Fuzz {
+
+DEFINE_FUZZER(const uint8_t* buf, size_t len) {
+  {
+    Envoy::Extensions::HttpFilters::CdnLoop::Parser::ParseContext input(
+        absl::string_view(reinterpret_cast<const char*>(buf), len));
+    Envoy::Extensions::HttpFilters::CdnLoop::Parser::parseCdnInfoList(input).IgnoreError();
```

Comment:
Very cool! I have a quick question -- looking at some of the unit tests, the ones that end up with OK status then check if the value contains expected CDN info. 

One way fuzzers can also detect errors on top of crashes is checking some expectation, e.g.
stringify(parse(input)) ==(sort of) input
parse(input).value().cdnIds() contains somePartOf(input)

Do you think anything like that applies in this case? I admit it seems difficult because of all the whitespace options, but curious to know if the successful outputs may have some structure behind it. (Or even lower level functions like parseToken)

Reply:
The only real structure is that the CDN ids we find should be in the input in the same order as the return value.  I've added code to do this.